snap/info_snap_yaml.go

```diff
@@ -324,6 +335,14 @@ func setAppsFromSnapYaml(y snapYaml, snap *Info) error {
 			app.Slots[slotName] = slot
 			slot.Apps[appName] = app
 		}
+		for name, data := range yApp.Sockets {
```

Comment:
The name needs to be validated as this is being used in multiple security sensitive places, including path names.

We can start reusing the ValidateName which validates snap names.

Reply:
Done (I actually added `ValidateAppSocketName` with the same logic, but different error message)